src/Operation/Find.php

```diff
@@ -179,6 +182,10 @@ public function __construct($databaseName, $collectionName, $filter, array $opti
             throw InvalidArgumentException::invalidType('"max" option', $options['max'], 'array or object');
         }
 
+        if (isset($options['maxAwaitTimeMS']) && ! is_integer($options['maxAwaitTimeMS'])) {
+            throw InvalidArgumentException::invalidType('"maxAwaitTimeMS" option', $options['maxAwaitTimeMS'], 'integer');
+        }
```

Comment:
Since this option has no effect unless `cursorType` is `TAILABLE_AWAIT`, we could consider adding one more check towards the end of the constructor. I usually do such multi-option checks for logical errors after all single option type checks (e.g. [FindAndModify](https://github.com/mongodb/mongo-php-library/blob/d10613869a820b1436ec27f3b99a61defe8f8886/src/Operation/FindAndModify.php#L156)).

That said, neither PHPC nor libmongoc enforce that. `maxAwaitTimeMS` will simply be ignored for cursors that aren't tailable and awaitData, so we're fine without it.